Sources/UInt128.swift

```diff
@@ -410,7 +410,7 @@ extension UInt128 : BinaryInteger {
 
             let masked = shifted & .init(upperBits: 0, lowerBits: mask)
 
-            words.append(UInt(masked.value.lowerBits))
+            words[idx] = UInt(masked.value.lowerBits)
```

Comment:
I prefer the append method here. I'm not a bit fan of using indexed arrays in modern programming languages. Do you have a particular reason to switch from appending to an empty array over to precreating the array size and then indexing into it?